File: me_daily/2025/10/1020/cf2148.java
import java.util.*;
import java.io.*;

public class cf2148 {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static PrintWriter pw = new PrintWriter(System.out);
    static StringTokenizer st;
    
    static String next() throws IOException {
        while (st == null || !st.hasMoreTokens()) {
            st = new StringTokenizer(br.readLine());
        }
        return st.nextToken();
    }
    
    static int nextInt() throws IOException {
        return Integer.parseInt(next());
    }
    
    static long nextLong() throws IOException {
        return Long.parseLong(next());
    }
    
    static long solve(long[] a) {
        List<Long> odds = new ArrayList<>();
        List<Long> evens = new ArrayList<>();
        long sumE = 0;
        
        for (long x : a) {
            if (x % 2 == 0) {
                evens.add(x);
                sumE += x;
            } else {
                odds.add(x);
            }
        }
        
        int k = odds.size();
        if (k == 0) return 0;
        
        Collections.sort(odds);
        
        long sumO = 0;
        for (long x : odds) {
            sumO += x;
        }
        
        int sac = k / 2;
        long sumSac = 0;
        for (int i = 0; i < sac; i++) {
            sumSac += odds.get(i);
        }
        
        return sumO - sumSac + sumE;
    }
    
    public static void main(String[] args) throws IOException {
        int t = nextInt();
        
        while (t-- > 0) {
            int n = nextInt();
            long[] a = new long[n];
            for (int i = 0; i < n; i++) {
                a[i] = nextLong();
            }
            
            long ans = solve(a);
            pw.println(ans);
        }
        
        pw.close();
    }
}

File: me_daily/2025/10/1020/cf2148.js
const readline = require('readline');
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

let lines = [];
let currentLine = 0;

function input() { return lines[currentLine++]; }
function nextInt() { return parseInt(input()); }
function nextIntArray() { return input().split(' ').map(Number); }

const INF = 0x3f3f3f3f;
const MOD = 1000000007;

function solve() {
    const t = nextInt();
    
    for (let test = 0; test < t; test++) {
        const n = nextInt();
        const a = nextIntArray();
        
        const odds = [];
        const evens = [];
        let sumE = 0;
        
        for (let x of a) {
            if (x % 2 === 0) {
                evens.push(x);
                sumE += x;
            } else {
                odds.push(x);
            }
        }
        
        const k = odds.length;
        if (k === 0) {
            console.log(0);
            continue;
        }
        
        odds.sort((a, b) => a - b);
        
        let sumO = 0;
        for (let x of odds) {
            sumO += x;
        }
        
        const sac = Math.floor(k / 2);
        let sumSac = 0;
        for (let i = 0; i < sac; i++) {
            sumSac += odds[i];
        }
        
        const ans = sumO - sumSac + sumE;
        console.log(ans);
    }
}

rl.on('line', (line) => { lines.push(line); });
rl.on('close', () => { solve(); });